services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/kafka/KafkaMessageProducer.java

```diff
@@ -99,30 +94,23 @@ public void postMessage(final String topic, final Message message) {
         heartBeat.reschedule();
     }
 
-    private void initProducer(Context context) {
-        if (!context.isTestingMode()) {
-            producer = new Producer(context);
+    private void initProducer(KafkaProducerConfig kafkaConfig, KafkaProducerConfig producerConfig) {
+        if (!producerConfig.isTestingMode()) {
+            producer = new Producer(kafkaConfig);
         } else {
-            producer = new TestAwareProducer(context);
+            producer = new TestAwareProducer(kafkaConfig);
         }
     }
 
-    private void initHeartBeat(Context context) throws FloodlightModuleException {
-        final String option = "heart-beat-interval";
-
-        String value = context.getHeartBeatInterval();
-
-        try {
-            Float interval = Float.valueOf(value);
-            if (interval < 1) {
-                throw new FloodlightModuleException(String.format(
-                        "Invalid value for option %s: %s < 1", option, value));
-            }
-            heartBeat = new HeartBeat(producer, (long) (interval * 1000), topoDiscoTopic);
-        } catch (NumberFormatException e) {
-            throw new FloodlightModuleException(String.format(
-                    "Invalid value for option %s=\"%s\", expect number", option, value));
+    private void initHeartBeat(KafkaProducerConfig producerConfig, String topoDiscoTopic)
+            throws FloodlightModuleException {
+        float interval = producerConfig.getHeartBeatInterval();
+        if (interval < 1) {
```

Comment:
Can/should it be validated on "config level"?

Reply:
Good to have it, but conf4j library doesn't provide any validation functionality. And such changes is out of the task scope, so I've introduced support of validation-api for configuration beans in another PR #1013